### ISSUE TITLE
Reconsider the "aborting" status

### ISSUE THREAD

**Issue by medihack** [procrastinate-org/procrastinate#1136]
It would be simpler if we had a dedicated field for an abortion request instead of an aborting status.

- The `aborting` state masks the `doing` state. State transitions from `aborting` are the same as from `doing`.
- Also, since the user does not necessarily have to follow this request with a dedicated field, it would still be preserved in a subsequent possible retry.
- Race conditions may occur that could solved with a dedicated field.

**Comment by medihack** [comment]
I already started working on a PR.

I added a boolean `abort` column to the jobs table. The migration will change all `aborting` states to `doing` states and set the `abort` to `true`.
One principle question: Should we keep the `abort` as `true` when the job is finished, or should we reset it to `false`? Does it matter somehow?

**Comment by medihack** [comment]
I came pretty far with everything (all but one test succeeded), but I think I need some help with the migration 😞. The migration is quite long as PostgreSQL does not allow the removal of one value from the enum type, so I have to create a new enum type and switch to it (and recreate all dependent indices, triggers, and functions). Also not a problem, but the problem is when changing the status column to the new type. Here is a reduced example:

```sql
-- Delete the triggers that depends on the old status type (to recreate them later)
DROP TRIGGER IF EXISTS procrastinate_jobs_notify_queue ON procrastinate_jobs;
DROP TRIGGER IF EXISTS procrastinate_trigger_status_events_update ON procrastinate_jobs;
DROP TRIGGER IF EXISTS procrastinate_trigger_status_events_insert ON procrastinate_jobs;
DROP TRIGGER IF EXISTS procrastinate_trigger_scheduled_events ON procrastinate_jobs;

-- Create a new enum type without 'aborting'
CREATE TYPE procrastinate_job_status_new AS ENUM (
    'todo',
    'doing',
    'succeeded',
    'failed',
    'cancelled',
    'aborted'
);

-- We need to drop the default temporarily as otherwise a DatatypeMismatch would occur
ALTER TABLE procrastinate_jobs ALTER COLUMN status DROP DEFAULT;

-- Alter the table to use the new type
ALTER TABLE procrastinate_jobs
ALTER COLUMN status TYPE procrastinate_job_status_new
USING (CASE
    WHEN status::text = 'aborting' THEN 'aborted'::procrastinate_job_status_new
    ELSE status::text::procrastinate_job_status_new
END);
```

And here the error I get because of the last statement:
```
2024-07-31 14:52:28.849 UTC [2525] ERROR:  operator does not exist: procrastinate_job_status_new = procrastinate_job_status at character 129
2024-07-31 14:52:28.849 UTC [2525] HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
2024-07-31 14:52:28.849 UTC [2525] STATEMENT:  -- Delete the triggers that depends on the old status type (to recreate them later)
```

I don't know how I could be even more explicit about the casting?!

**Comment by ewjoachim** [comment]
> Should we keep the abort as true when the job is finished, or should we reset it to false? Does it matter somehow?

(As long as the job is not retried, I don't have a strong opinion)

**Comment by ewjoachim** [comment]
Fixed by #1139
